src/contrib/persistence/Akka.Persistence.Sql.Common/Journal/JournalDbEngine.cs

```diff
@@ -211,39 +212,58 @@ void IDisposable.Dispose()
         /// Specific table used for message persistence may be defined through configuration within 
         /// 'akka.persistence.journal.sql-server' scope with 'schema-name' and 'table-name' keys.
         /// </summary>
-        public async Task WriteMessagesAsync(IEnumerable<IPersistentRepresentation> messages)
+        public async Task<IImmutableList<Exception>> WriteMessagesAsync(IEnumerable<AtomicWrite> messages)
```

Comment:
Actually, on second thought, IEnumerable is immutable, but the underlying implementation is not guaranteed to be, so it's possible for the creator of the IEnumerable to make changes after passing the reference back. Should we ask for a second opinion @Horusiath? :)